src/components/views/dialogs/SetDisplayNameDialog.js

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+var React = require("react");
+var sdk = require("../../../index.js");
+var MatrixClientPeg = require("../../../MatrixClientPeg");
+
+module.exports = React.createClass({
+    displayName: 'SetDisplayNameDialog',
+    propTypes: {
+        onFinished: React.PropTypes.func.isRequired,
+        currentDisplayName: React.PropTypes.string,
+    },
+
+    getInitialState: function() {
+        return {
+            value: this.props.currentDisplayName || this.getDefaultDisplayName(),
+        }
+    },
+
+    getDefaultDisplayName: function() {
```

Comment:
Any reason why we're increasing our public API surface like this, rather than just inlining this on `getInitialState`?